vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -363,20 +363,20 @@ func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isRead
 	return mountpoint, fs.MountWithID(mountpoint, fstype, id, isReadOnly)
 }
 
+//VolumesInRefMap - get list of volumes from refmap
+func (d *VolumeDriver) VolumesInRefMap() []string {
+	return d.refCounts.GetVolumeNames()
+}
+
 // private function that does the job of mounting volume in conjunction with refcounting
 func (d *VolumeDriver) processMount(r volume.MountRequest) volume.Response {
-	// get volume metadata
-	volumeMeta, err := d.GetVolume(r.Name)
-
-	if err != nil {
-		return volume.Response{Err: err.Error()}
-	}
-
-	r.Name, err = plugin_utils.GetFullVolumeName(r.Name, volumeMeta["datastore"].(string), d)
+	volname, volumeMeta, err := plugin_utils.GetFullNameAndMeta(r.Name, "", d)
 	if err != nil {
 		log.Errorf("Unable to get full name for volume %s. err:%v", r.Name, err)
 		return volume.Response{Err: err.Error()}
 	}
+	r.Name = volname
+	d.mountIDtoName[r.ID] = volname
```

Comment:
nit - I'd use r.Name here, not volname (just like everywhere till the rest of the function)